backend/app/api/routes/document.py

```diff
@@ -0,0 +1,35 @@
+from fastapi import FastAPI, HTTPException, APIRouter
+from fastapi.responses import StreamingResponse
+from sqlmodel import Session
+from app.api.deps import SessionDep
+from app.repositories import document_repo
+from app.file_storage import get_file_storage
+
+router = APIRouter()
+
+@router.get("/documents/{doc_id}/download")
+def download_file(
+    doc_id: int,
+    session: SessionDep
+):
+    doc = document_repo.must_get(session, doc_id)
+
+    name = doc.source_uri
+    filestorage = get_file_storage()
+    if filestorage.exists(name):
+        file_size = filestorage.size(name)
+        headers = {"Content-Length": str(file_size)}
+        def iterfile():
+            with filestorage.open(name) as f:
+                while chunk := f.read(8192):  # 每次读取 8KB
```

Comment:
Please refer to [FastAPI Document](https://fastapi.tiangolo.com/advanced/custom-response/#using-streamingresponse-with-file-like-objects). It's not necessary to manually call `f.read()`.

Reply:
done